index.js

```diff
@@ -307,24 +372,44 @@
         return f('(' + stripNamespace(name) + ' ') +
                f$1(String($1)) + f(' ') + f$2(String($2)) + f(')');
       };
-      return {
+      var validate = function(x) {
+        if (!test(x)) {
+          return Left({value: x, typePath: [t], propPath: []});
+        }
+        for (var n = 1; n <= 2; n += 1) {
+          var _ = '_' + String(n);
+          var $ = '$' + String(n);
+          for (var idx = 0, xs = t[_](x); idx < xs.length; idx += 1) {
+            var result = t[$].validate(xs[idx]);
+            if (result.isLeft) {
+              return Left({value: result.value.value,
```

Comment:
```haskell
result :: Either { value :: Any, typePath: [Type], propPath: [String] } Any
```

The first `.value` provides access to the Left's value, which happens to be a record with a `value` field. ;)